src/cpp/ext/proto_server_reflection.cc

```diff
@@ -117,9 +117,9 @@ Status ProtoServerReflection::ListService(ServerContext* context,
   if (services_ == nullptr) {
     return Status(StatusCode::NOT_FOUND, "Services not found.");
   }
-  for (auto it = services_->begin(); it != services_->end(); ++it) {
+  for (auto value : services_) {
```

Comment:
The semantics of the old version are best maintained, IMO, if this is `const auto& value` , since I'm not sure if the compiler will automatically set this otherwise.